lib/elixir/lib/calendar/iso.ex

```diff
@@ -447,4 +461,136 @@ defmodule Calendar.ISO do
       {days + days_offset, {parts, ppd}}
     end
   end
+
+  # :calendar.date_to_gregorian_days(year, month, day) - 365
+  # `year >=0`-check only exists to maintain full compatibility with `:calendar`,
+  # Can be removed in the future.
+  defp date_to_gregorian_days(year, month, day) when year >= 0 do
+    last_day = days_in_month(year, month)
+    true =  day <= last_day
```

Comment:
Shall we keep the assertion or remove it? 

Personally I think it is not necessary as it is an internal function, but please tell me your opinion.